### ISSUE TITLE
[GlobalStep] Blank Screen is displayed after clicking "Woocommerce"> "Home"

### ISSUE THREAD

**Issue by gglobalstep** [Automattic/woocommerce-payments#3333]
# Please do not close this issue before the changes [are re applied](https://github.com/Automattic/woocommerce-payments/issues/3333#issuecomment-972113433)

# Bug Description:
Blank Screen is displayed after clicking "Woocommerce"> "Home"

# Environment:
Woocommerce Version : WooCommerce  5.9.0 
WooCommerce Payments - 3.3.0-test-2
WordPress version: v5.8.1

PC:
Windows 10
Chrome(Version 95.0.4638.69)
Safari: v14.0

# Steps To Reproduce:
1. Create any test site using JN site.
1. Install and activate all the required plugins.
1. Install and activate  WooCommerce Payments - 3.3.0-test-2.
1. Click on woocommerce home page.
1. Observe Blank Screen.

# Actual Result:
Blank Screen is displayed after clicking "Woocommerce"> "Home"

# Expected Result:
After clicking "woocommerce home page"appropriate page should be displayed.

# Attachments:

https://user-images.githubusercontent.com/41110392/140975305-72225c9f-c9c5-44d6-bb59-ad2222d5dbbf.mp4

*Isolating the problem (mark completed items with an [x]):*
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

<details>
`
### WordPress Environment ###

WC Version: 5.9.0
REST API Version: ✔ 5.9.0
WC Blocks Version: ✔ 6.2.0
Action Scheduler Version: ✔ 3.3.0
WC Admin Version: ✔ 2.8.0
Log Directory Writable: ✔
WP Version: 5.8.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.51 (Unix) OpenSSL/1.0.2g
PHP Version: 7.4.25
PHP Post Max Size: 1 GB
PHP Time Limit: 30
PHP Max Input Vars: 5000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
MySQL Version: 5.7.33-0ubuntu0.16.04.1-log
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.9.0
WC Database Prefix: wp_
Total Database Size: 5.35MB
Database Data Size: 3.75MB
Database Index Size: 1.60MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.09MB + Index: 0.13MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.06MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 2.41MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.23MB + Index: 0.20MB + Engine InnoDB
wp_posts: Data: 0.08MB + Index: 0.06MB + Engine InnoDB
wp_snippets: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 24
page: 9
post: 1
product: 21
product_variation: 7
revision: 2
shop_order: 28
shop_order_refund: 4
shop_subscription: 8

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (7) ###

Query Monitor: by John Blackbourn – 3.7.1
Code Snippets: by Code Snippets Pro – 2.14.2
Companion Plugin: by Osk – 1.18
WooCommerce Blocks: by Automattic – 6.2.0
WooCommerce Payments Dev Tools: by Automattic –
WooCommerce Payments: by Automattic – 3.3.0-test-2
WooCommerce: by Automattic – 5.9.0

### Inactive Plugins (3) ###

Akismet Anti-Spam: by Automattic – 4.1.12
Hello Dolly: by Matt Mullenweg – 1.7.2
WooCommerce Subscriptions: by WooCommerce – 3.1.6

### Dropin Plugins (1) ###

db.php: Query Monitor Database Class

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/
Checkout: #7 - /checkout/
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 3.9.1
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Subscriptions ###

WCS_DEBUG: ✔ No
Subscriptions Mode: ✔ Live
Subscriptions Live URL: https://ideal-cockroach.jurassic.ninja
Subscription Statuses: wc-active: 8
WooCommerce Account Connected: ❌ No

### Store Setup ###

Country / State: United States (US) — California

### Subscriptions by Payment Gateway ###

WooCommerce Payments: wc-active: 8

### Payment Gateway Support ###

WooCommerce Payments: products
refunds
subscriptions
subscription_cancellation
subscription_suspension
subscription_reactivation
subscription_amount_changes
subscription_date_changes
subscription_payment_method_change
subscription_payment_method_change_customer
subscription_payment_method_change_admin
multiple_subscriptions
tokenization
add_payment_method

### WooCommerce Payments ###

Version: 3.3.0-test-2
Connected to WPCOM: Yes
Blog ID: 199647562
Account ID: acct_1Jtp1V2HFljPTaqo

### Action Scheduler ###

Complete: 182
Oldest: 2021-11-09 08:03:05 +0000
Newest: 2021-11-09 11:43:03 +0000

Pending: 12
Oldest: 2021-11-09 18:39:41 +0000
Newest: 2022-01-09 11:47:02 +0000

### Status report information ###

Generated at: 2021-11-09 16:46:23 +00:00

</details>

**Comment by jrodger** [comment]
I've replicated the issue. It doesn't happen on my existing store, so it could be related to the state of the task list on the WooCommerce home screen?

There's more going on here than I've investigated, but it doesn't happen on WCPay 3.2.3.

My current theory is some interplay between WooCommerce, WooCommerce Admin, and WooCommerce Payments.

**Comment by jrodger** [comment]
Console output:

```
data.min.js?ver=a9e98fe3ad76884fe7909ba8348395b7:2 An error occurred while running 'mapSelect': o is not a function
Te @ data.min.js?ver=a9e98fe3ad76884fe7909ba8348395b7:2
react-dom.min.js?ver=16.13.1:125 TypeError: Cannot destructure property 'trackedCompletedTasks' of 'Object(...)(...)' as it is undefined.
    at t.default (task-list.js?ver=2.8.0:1)
    at we (react-dom.min.js?ver=16.13.1:84)
    at He (react-dom.min.js?ver=16.13.1:97)
    at zj (react-dom.min.js?ver=16.13.1:228)
    at Th (react-dom.min.js?ver=16.13.1:152)
    at tj (react-dom.min.js?ver=16.13.1:152)
    at Te (react-dom.min.js?ver=16.13.1:146)
    at react-dom.min.js?ver=16.13.1:61
    at unstable_runWithPriority (react.min.js?ver=16.13.1:25)
    at Da (react-dom.min.js?ver=16.13.1:60)
Me @ react-dom.min.js?ver=16.13.1:125
react.min.js?ver=16.13.1:24 Uncaught TypeError: Cannot destructure property 'trackedCompletedTasks' of 'Object(...)(...)' as it is undefined.
    at t.default (task-list.js?ver=2.8.0:1)
    at we (react-dom.min.js?ver=16.13.1:84)
    at He (react-dom.min.js?ver=16.13.1:97)
    at zj (react-dom.min.js?ver=16.13.1:228)
    at Th (react-dom.min.js?ver=16.13.1:152)
    at tj (react-dom.min.js?ver=16.13.1:152)
    at Te (react-dom.min.js?ver=16.13.1:146)
    at react-dom.min.js?ver=16.13.1:61
    at unstable_runWithPriority (react.min.js?ver=16.13.1:25)
    at Da (react-dom.min.js?ver=16.13.1:60)
t.default @ task-list.js?ver=2.8.0:1
we @ react-dom.min.js?ver=16.13.1:84
He @ react-dom.min.js?ver=16.13.1:97
zj @ react-dom.min.js?ver=16.13.1:228
Th @ react-dom.min.js?ver=16.13.1:152
tj @ react-dom.min.js?ver=16.13.1:152
Te @ react-dom.min.js?ver=16.13.1:146
(anonymous) @ react-dom.min.js?ver=16.13.1:61
unstable_runWithPriority @ react.min.js?ver=16.13.1:25
Da @ react-dom.min.js?ver=16.13.1:60
Pg @ react-dom.min.js?ver=16.13.1:61
U @ react.min.js?ver=16.13.1:16
B.port1.onmessage @ react.min.js?ver=16.13.1:24
VM1269:1 POST https://relative-cow.jurassic.ninja/wp-admin/admin-ajax.php 503
(anonymous) @ VM1269:1
send @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
ajax @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
s.ajax.s.ajax @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:5
l @ heartbeat.min.js?ver=5.8.1:2
(anonymous) @ heartbeat.min.js?ver=5.8.1:2
setTimeout (async)
d @ heartbeat.min.js?ver=5.8.1:2
(anonymous) @ heartbeat.min.js?ver=5.8.1:2
c @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
fireWith @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
l @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
(anonymous) @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
load (async)
send @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
ajax @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
s.ajax.s.ajax @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:5
l @ heartbeat.min.js?ver=5.8.1:2
d @ heartbeat.min.js?ver=5.8.1:2
v @ heartbeat.min.js?ver=5.8.1:2
(anonymous) @ heartbeat.min.js?ver=5.8.1:2
dispatch @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
v.handle @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
VM1269:1 POST https://relative-cow.jurassic.ninja/wp-admin/admin-ajax.php 503
(anonymous) @ VM1269:1
send @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
ajax @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
s.ajax.s.ajax @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:5
l @ heartbeat.min.js?ver=5.8.1:2
d @ heartbeat.min.js?ver=5.8.1:2
v @ heartbeat.min.js?ver=5.8.1:2
(anonymous) @ heartbeat.min.js?ver=5.8.1:2
dispatch @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
v.handle @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=5.8.1:2
```

**Comment by asumaran** [comment]
> It doesn't happen on my existing store

I got the same issue after updating WooCommerce to 5.9.0 on my existing WCPay installation and also in a fresh install. The error only appears when activating WCPay (`develop`).

**Comment by leonardola** [comment]
I am taking a look at that. I can replicate locally but only with the latest WooCommerce version. Reverting my changes seem to fix the bug. It seems like there is some type of integration problem between my changes and the latest WooCommerce version

**Comment by leonardola** [comment]
Tested again with a fresh install with WooCommerce 5.8 and WCPay 3.3.0-test-2 works.
updating to WooCommerce 5.9 breaks

**Comment by leonardola** [comment]
Reverting only the ExperimentalList change does not fix the problem. Only a full downgrade fixes it.

**Comment by leonardola** [comment]
This bug was introduced in version 6.1.2 of @woocommerce/components
The only differences between 6.1.1 to 6.1.2 are inside the file package.json :
<img width="1539" alt="Screen Shot 2021-11-11 at 5 58 37 PM" src="https://user-images.githubusercontent.com/6342964/141368144-1cc7903d-fa32-4355-80ea-1fea1b1c1849.png">

**Comment by leonardola** [comment]
PR reverting the changes temporarily created [here](https://github.com/Automattic/woocommerce-payments/pull/3370)


As the bug started after a change that makes `@woocommerce/components` use some packages bundled inside WooCommerce plugin and only after the latest WooCommerce release(5.9.0). The bug seems to actually be inside the packages bundled by WooCommerce not inside WCPay. @jrodger should I keep looking for the bug or should we port this issue to WooCommerce repository?

The error happens [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/data/src/components/use-select/index.js#L139). Apparently the registry is not being created after the last WooCommerce Update

**Comment by jrodger** [comment]
> should I keep looking for a fix or should we port this issue to WooCommerce repository?

It sounds like WCPay's usage of the WooCommerce Admin components is correct, so creating an issue in either the WC Admin or WooCommerce repository makes sense to me. I'm not sure which would be most appropriate, perhaps WC Admin based on my understanding of the problem?

Rather than porting this issue across, would you mind creating a new issue and summarising your investigation so far? I think that would be more useful then someone trying to understand the complete history.

**Comment by leonardola** [comment]
Issue created on WooCommerce repository [here](https://github.com/woocommerce/woocommerce/issues/31241). Waiting on a fix to be able to redo the changes of this [PR](https://github.com/Automattic/woocommerce-payments/pull/3370)

**Comment by vbelolapotkov** [comment]
Marking as blocked for now.

**Comment by naman03malhotra** [comment]
GlobalStep is not able to reproduce this issue, closing it for now ref: pbPjnJ-Ps-p2#comment-729
